src/scheduler/JobScheduler.ts

```diff
@@ -149,23 +151,57 @@ export class JobScheduler {
           : jobEntity.concurrency;
       this.logger.debug('execute job', { name: this.jobName, times: numToExecute });
 
-      for (let i = 0; i < numToExecute; i++) {
+      this.executorError = new Array(numToExecute);
+
+      for (let instanceNumber = 0; instanceNumber < numToExecute; instanceNumber++) {
+        this.logger.debug('executing job instance', { instanceNumber });
+        this.executorError[instanceNumber] = false;
+
+        const executionTimeout =
+          jobEntity.timeout !== undefined
+            ? setSafeTimeout(
+                async () => this.handleTimeoutReached(instanceNumber),
+                jobEntity.timeout,
+                this.logger,
+                'error occurred in timeout',
+              )
+            : undefined;
+
         // eslint-disable-next-line no-void
-        void this.jobExecutor.execute(jobEntity, parameters).catch((e) => {
-          this.handleUnexpectedError(e);
-        });
+        void this.jobExecutor
+          .execute(jobEntity, parameters)
+          .catch((e) => {
+            this.handleUnexpectedError(instanceNumber, e);
+          })
+          .finally(() => {
+            if (this.executorError[instanceNumber] !== undefined) {
```

Comment:
I don´t get this, this will be true or false, why check for undefined? Wont that always evaluate to `true`?